src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -360,12 +360,15 @@ class Crystal::Doc::Generator
     filename[@base_dir.size..-1]
   end
 
-  record RelativeLocation, filename : String, line_number : Int32, url : String? do
+  record RelativeLocation, filename : String, line_number : Int32, url : String?, show_line_number : Bool do
```

Comment:
`show_line_number` -> `show_line_number?`

Reply:
I'm not sure if this is a better name for `show_line_number`. You normally use this `?` at the end of the name of Bool methods where you are checking something (like `uppercase?`). Or do you mean `show_line_number : Bool?`?

Reply:
Well, it **is** a `Bool` returning method, quite query-ish (`show_...`) for my taste.

Reply:
`@show_line_number?` is not even a valid name for an instance variable.

Reply:
Sorry :/ @straight-shoota No, but it's still a valid name for an accessor.

Reply:
Sorry, but `show_line_number` is perfectly clear and fine. There's no improvement at all to use `?` at the end here, it's not like this is an API that's used all over the place, and adding that `?` will mean defining the getter in a block, which will make the code less clear.

Reply:
> No, but it's still a valid name for an accessor.

You can't use it in the record definition. Which means you'd need to declare an additional getter in the record body. Not worth it. It's just a simple data struct.